zkvm/src/types.rs

```diff
@@ -277,19 +280,24 @@ impl Item {
 }
 
 impl Data {
-    /// Returns the length of the underlying vector of bytes.
-    pub fn len(&self) -> usize {
+    /// Returns a guaranteed lower bound on the number of bytes
+    /// needed to serialize the Data.
+    pub fn min_serialized_length(&self) -> usize {
         match self {
             Data::Opaque(data) => data.len(),
-            Data::Witness(_) => unimplemented!(),
+            Data::Witness(x) => 0,
         }
     }
 
     /// Converts the Data into a vector of bytes
     pub fn to_bytes(self) -> Vec<u8> {
         match self {
             Data::Opaque(data) => data,
-            Data::Witness(_) => unimplemented!(),
+            Data::Witness(w) => {
+                let mut bytes: Vec<u8> = Vec::new();
```

Comment:
maybe, so later we only have to improve this in one place?

```suggestion
                let mut bytes: Vec<u8> = Vec::with_capacity(self.min_serialized_length());
```